src/wombats/sockets/game.clj

```diff
@@ -192,64 +192,77 @@
 
 (defn- build-simulation-game-state
   [socket-user wombat-id simulator-template-id datomic]
-  (let [simulator-template
-        ((:get-simulator-arena-template-by-id datomic)
-         simulator-template-id)
-
-        user
-        ((:get-entire-user-by-id datomic)
-         (:user/id socket-user))
-
-        wombat
-        ((:get-wombat-by-id datomic)
-         wombat-id)]
-
-    {:arena-config (:simulator-template/arena-template simulator-template)
-     :players {(gen-id) {:player {:player/color "gray"}
-                         :stats initial-stats
-                         :user {:user/github-username (:user/github-username user)
-                                :user/github-access-token (:user/github-access-token user)}
-                         :wombat {:wombat/id (:wombat/id wombat)
-                                  :wombat/name (:wombat/name wombat)
-                                  :wombat/url (:wombat/url wombat)}
-                         :state {:code nil
-                                 :command nil
-                                 :error nil
-                                 :saved-state {}}}}
-     :frame {:frame/frame-number 0
-             :frame/round-number 1
-             :frame/round-start-time nil
-             :frame/arena (:simulator-template/arena simulator-template)}}))
-
-(defn t-sim
-  [game-state]
-  (doseq [[_ zakano] (:zakano game-state)]
-    (prn (str (get-in zakano [:state :code :code]))))
-  game-state)
+  (try
+    (let [simulator-template
+          ((:get-simulator-arena-template-by-id datomic)
+           simulator-template-id)
+
+          user
+          ((:get-entire-user-by-id datomic)
+           (:user/id socket-user))
+
+          wombat
+          ((:get-wombat-by-id datomic)
+           wombat-id)]
+
+      (when-not simulator-template
+        (throw (Exception. "no simulator template found")))
+
+      (when-not user
+        (throw (Exception. "no user found")))
+
+      (when-not wombat
+        (throw (Exception. "no wombat found")))
+
+      {:arena-config (:simulator-template/arena-template simulator-template)
+       :players {(gen-id) {:player {:player/color "gray"}
+                           :stats initial-stats
+                           :user {:user/github-username (:user/github-username user)
+                                  :user/github-access-token (:user/github-access-token user)}
+                           :wombat {:wombat/id (:wombat/id wombat)
+                                    :wombat/name (:wombat/name wombat)
+                                    :wombat/url (:wombat/url wombat)}
+                           :state {:code nil
+                                   :command nil
+                                   :error nil
+                                   :saved-state {}}}}
+       :frame {:frame/frame-number 0
```

Comment:
So I think it depends. Were using the `:frame/` namespaced keywords because that's how they are stored in datomic. I'm using the `:frame` property in the map just to segment the frame data. Not sure if there is a better way to leverage the namespaced keywords to remove the redundancy.